src/simulator_core/entities/assets/demand_cluster.py

```diff
@@ -48,15 +48,15 @@ def add_physical_data(self, data: Dict[str, float]) -> None:
         """
         pass
 
-    def set_setpoints(self, setpoints: Dict, **kwargs) -> None:
+    def set_setpoints(self, setpoints: Dict, **kwargs: Dict) -> None:
```

Comment:
Do you need a :Dict definition when you add keyword arguments. These are already in a dictionary format because they are keyword arguments?

Reply:
I added this based on type checking issue. I agree it feels a bit weird, but then we need to adjust type checking to not give error on this.

Reply:
@MichielTukker  what's your opinion?